### DISCUSSION TITLE
Proventos

### DISCUSSION THREAD

**Discussion by dzambelle** [gicornachini/bolsa#6] [Ideas]
Fala Giovanni, tudob bem?
senti falta apenas na coleta do proventos (anuncio de proventos) no site do CEI (investimentos/proventos)

tem alguma idéia de adicionar isso ao código?

abs

**Comment by gicornachini** [comment]
A princípio não teria essa ideia. Acho muito bacana adicionar uma funcionalidade de informar proventos, mas eu não tenho certeza se as informações que são exibidas no cei são suficientes para chegar no resultado que é esperado: exibir todos os proventos recebidos. Dei uma rápida analisada e parece que o único [local que temos essa informação](https://ceiapp.b3.com.br/CEI_Responsivo/ConsultarProventos.aspx) é bem limitada por período...

**Comment by dzambelle** [comment]
Sim. Pensei em fazer um batch de consulta diaria e alimentar um BD

Mas confesso que estou iniciando em python.

Vou te avisando conforme evolua.

Abs e obrigado

Enviado do meu Galaxy



-------- Mensagem original --------
De : Giovanni Cornachini ***@***.***>
Data: 12/05/2021 18:53 (GMT-03:00)
Para: gicornachini/bolsa ***@***.***>
Cc: dzambelle ***@***.***>, Author ***@***.***>
Assunto: Re: [gicornachini/bolsa] Proventos (#6)


A princípio não teria essa ideia. Acho muito bacana adicionar uma funcionalidade de informar proventos, mas eu não tenho certeza se as informações que são exibidas no cei são suficientes para chegar no resultado que é esperado: exibir todos os proventos recebidos. Dei uma rápida analisada e parece que o único local que temos essa informação<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fceiapp.b3.com.br%2FCEI_Responsivo%2FConsultarProventos.aspx&data=04%7C01%7C%7Cae57bd7124ed454ccbf908d915905260%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637564531862749031%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=cpk0Mc8sb7CjtNLO6%2FMRscJmwP4kCduoaBAiF2Q8Ec0%3D&reserved=0> é bem limitada por período...

**Comment by dzambelle** [comment]
Fala Giovanni,

Terminei de implementar no seu codigo a consulta também aos proventos
Realmente as datas sao limitadas, mas executando diáriamente não tem problema.

Me avisa qual a melhor forma de eu compartilhar com você o código (não sei ainda usar muito bem o git)

**Reply by gicornachini** [reply]
Eu acredito que a solução de adicionar um BD no projeto se torna inviável por questão de versionamento no código/projeto e propósito do projeto(que no caso é um http client para obter dados do cei). Mas não vejo problema em ter a opção de um método para obter os dados de proventos(sem persistência local). Para contribuir com o projeto, o ideal é você fazer um `fork` do projeto e abrir um Pull Request com a sua proposta de solução! O github tem algumas docs que podem te ajudar nisso(além de vídeos no youtube): https://docs.github.com/pt/github/collaborating-with-issues-and-pull-requests/proposing-changes-to-your-work-with-pull-requests

**Reply by wagnervielmond** [reply]
Boa noite. Estou usando esse projeto para baixar dados do CEI, está funcionando muito bem, porém gostaria de saber se vocês podem compartilhar a parte do código que consulta os proventos? Agradeceria demais.

**Reply by wagnervielmond** [reply]
Sim, está funcionando, porém vc deve ter a versão do Python 3.9. Tbm criei a o script pra baixar os proventos: https://github.com/wagnervielmond/ProventosB3 usando o script que eles usaram.

**Reply by erivan-as** [reply]
Vocês conseguem fazer login no CEI utlizando esta biblioteca? Eu não consigo
